mlrun/model.py

```diff
@@ -577,6 +679,103 @@ def handler_name(self):
+            cleared_inputs[input_key] = input_value
+
+        # Set the now configuration free inputs:
+        self.inputs = cleared_inputs
+
+    @staticmethod
+    def join_outputs_and_returns(
+        outputs: List[str], returns: List[Union[str, Dict[str, str]]]
+    ) -> List[str]:
+        """
+        Get the outputs set in the spec. The outputs are constructed from both the 'outputs' and 'returns' properties
+        that were set by the user.
+
+        :param outputs: A spec outputs property - list of output keys.
+        :param returns: A spec returns property - list of key and configuration of how to log returning values.
+
+        :return: The joined 'outputs' and 'returns' list.
+        """
+        # Collect the 'returns' property keys:
+        cleared_returns = []
+        if returns:
+            for return_value in returns:
+                # Check if the return entry is a configuration dictionary or a key-type structure string (otherwise its
+                # just a key string):
+                if isinstance(return_value, dict):
+                    # Set it to the artifact key:
+                    return_value = return_value["key"]
+                elif ":" in return_value:
+                    # Take only the key name:
+                    return_value, _ = RunSpec._separate_by_colon(value=return_value)
+                # Collect it:
+                cleared_returns.append(return_value)
+
+        # Use `set` join to combine the two lists without duplicates:
+        outputs = list(set(outputs if outputs else []) | set(cleared_returns))
+
+        return outputs
+
+    @staticmethod
+    def _separate_by_colon(value: str) -> Tuple[str, str]:
+        """
+        An input key in the `inputs` dictionary parameter of a task (or `RunTime.run` method) or the docs setting of a
+        `RunTime` handler can be provided with a colon to specify its type hint in the following structure:
+        "<parameter_key> : <type_hint>".
+
+        Same for a returning output entry in the `returns` list parameter can be provided with a colon to specify its
+        artifact type to be logged as in the following structure:
+        "<artifact_key> : <artifact_type>".
+
+        This method parses the provided value by the user.
+
+        :param value: A string entry in the inputs dictionary keys or the returns list.
+
+        :return: The value as key and type tuple.
+
+        :raise MLRunInvalidArgumentError: If an incorrect pattern was provided.
+        """
+        # Validate correct pattern:
+        if value.count(":") > 1:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Incorrect input / return pattern. Inputs keys can have only a single ':' in them to specify the "
+                f"desired type the input will be parsed as. The same applies for returning outputs keys - only a "
+                f"single ':' to specify the artifact type. Given: {value}"
+            )
```

Comment:
Instead of confusing the user both with returns and inputs, because you know where you call each function, maybe pass to the method also the value type AKA inputs or outputs, that way you can have more clear error message raised to the user.

Reply:
done.